templates/terraform/examples/route_ilb_beta.tf.erb

```diff
@@ -0,0 +1,52 @@
+resource "google_compute_network" "default" {
+  provider      = "google-beta"
+  name = "<%= ctx[:vars]['network_name'] %>"
+  auto_create_subnetworks = false
+}
+
+resource "google_compute_subnetwork" "default" {
+  provider      = "google-beta"
+  name          = "<%= ctx[:vars]['subnet_name'] %>"
+  ip_cidr_range = "10.0.1.0/24"
+  region        = "us-central1"
+  network       = "${google_compute_network.default.self_link}"
+}
+
+resource "google_compute_health_check" "hc" {
+  provider           = "google-beta"
+  name               = "<%= ctx[:vars]['health_check_name'] %>"
+  check_interval_sec = 1
+  timeout_sec        = 1
+
+  tcp_health_check {
+    port = "80"
+  }
+}
+
+resource "google_compute_region_backend_service" "backend" {
+  provider              = "google-beta"
+  name                  = "<%= ctx[:vars]['backend_name'] %>"
+  region                = "us-central1"
+  health_checks         = ["${google_compute_health_check.hc.self_link}"]
+}
+
+resource "google_compute_forwarding_rule" "default" {
+  provider              = "google-beta"
+  name                  = "<%= ctx[:vars]['forwarding_rule_name'] %>"
+  region                = "us-central1"
+
+  load_balancing_scheme = "INTERNAL"
+  backend_service       = "${google_compute_region_backend_service.backend.self_link}"
+  all_ports             = true
+  network               = "${google_compute_network.default.name}"
+  subnetwork            = "${google_compute_subnetwork.default.name}"
+}
+
+resource "google_compute_route" "default" {
```

Comment:
`primary_resource_id` is supposed to be used within the example template to allow terraform to access the resource's id for import test purposes. It doesn't get used in beta-only examples, but can you change it so that when this moves to the `google` provider it will work?

Example: https://github.com/GoogleCloudPlatform/magic-modules/blob/master/templates/terraform/examples/node_template_server_binding.tf.erb

Reply:
ya i have added primary id

Reply:
Can you replace `"google_compute_route" "default"` with `"google_compute_route" "<%= ctx[:primary_resource_id] %>"` here

It allows us to access the name of the terraform resource for import later. This is different than the `name` field on the resource itself, as we need to know the name used to identify the block for testing import